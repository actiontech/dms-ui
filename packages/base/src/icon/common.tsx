export const IconCommonPassword: React.FC = () => {
  return (
    <svg
      width="18"
      height="19"
      viewBox="0 0 18 19"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g id="lock-password-fill">
        <path
          id="Vector"
          d="M13.5 6.5H15C15.1989 6.5 15.3897 6.57902 15.5303 6.71967C15.671 6.86032 15.75 7.05109 15.75 7.25V16.25C15.75 16.4489 15.671 16.6397 15.5303 16.7803C15.3897 16.921 15.1989 17 15 17H3C2.80109 17 2.61032 16.921 2.46967 16.7803C2.32902 16.6397 2.25 16.4489 2.25 16.25V7.25C2.25 7.05109 2.32902 6.86032 2.46967 6.71967C2.61032 6.57902 2.80109 6.5 3 6.5H4.5V5.75C4.5 4.55653 4.97411 3.41193 5.81802 2.56802C6.66193 1.72411 7.80653 1.25 9 1.25C10.1935 1.25 11.3381 1.72411 12.182 2.56802C13.0259 3.41193 13.5 4.55653 13.5 5.75V6.5ZM12 6.5V5.75C12 4.95435 11.6839 4.19129 11.1213 3.62868C10.5587 3.06607 9.79565 2.75 9 2.75C8.20435 2.75 7.44129 3.06607 6.87868 3.62868C6.31607 4.19129 6 4.95435 6 5.75V6.5H12ZM8.25 11V12.5H9.75V11H8.25ZM5.25 11V12.5H6.75V11H5.25ZM11.25 11V12.5H12.75V11H11.25Z"
          fill="#C3C6CD"
        />
      </g>
    </svg>
  );
};

export const IconCommonUser: React.FC = () => {
  return (
    <svg
      width="18"
      height="19"
      viewBox="0 0 18 19"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g id="user-4-fill">
        <g id="Vector">
          <path
            d="M6.47462 7.34681C5.80485 6.67704 5.42857 5.76863 5.42857 4.82143C5.42857 3.87423 5.80485 2.96582 6.47462 2.29605C7.14439 1.62627 8.0528 1.25 9 1.25C9.9472 1.25 10.8556 1.62627 11.5254 2.29605C12.1952 2.96582 12.5714 3.87423 12.5714 4.82143C12.5714 5.76863 12.1952 6.67704 11.5254 7.34681C10.8556 8.01658 9.9472 8.39286 9 8.39286C8.0528 8.39286 7.14439 8.01658 6.47462 7.34681Z"
            fill="#C3C6CD"
          />
          <path
            d="M16.5 15.1785C16.5 12.0225 13.9418 9.46426 10.7857 9.46426H7.21429C4.05821 9.46426 1.5 12.0225 1.5 15.1785V15.5357C1.5 15.9146 1.65051 16.2779 1.91842 16.5458C2.18633 16.8137 2.54969 16.9643 2.92857 16.9643H15.0714C15.4503 16.9643 15.8137 16.8137 16.0816 16.5458C16.3495 16.2779 16.5 15.9146 16.5 15.5357V15.1785Z"
            fill="#C3C6CD"
          />
        </g>
      </g>
    </svg>
  );
};
