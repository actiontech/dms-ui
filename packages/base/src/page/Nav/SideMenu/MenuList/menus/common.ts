import { UserRolesType } from '@actiontech/shared/lib/enum';

import {
  CustomMenuItemType,
  GenerateMenuItemType,
  MenuStructTreeKey,
  MenuStructTreeType
} from './index.type';

export const SIDE_MENU_DATA_PLACEHOLDER_KEY = 'projectID';

export const genMenuItemsWithMenuStructTree = (
  projectID: string,
  allMenuItems: GenerateMenuItemType[],
  menuStructTree: MenuStructTreeType,
  userRoles: UserRolesType
): CustomMenuItemType[] => {
  const getMenuItemWithKey = (key: MenuStructTreeKey): CustomMenuItemType => {
    return (
      allMenuItems.find((v) => {
        const menu = v(projectID);
        if (menu?.role) {
          return (
            menu.role.some((role) => userRoles[role]) && menu.structKey === key
          );
        }

        return menu?.structKey === key;
      })?.(projectID) ?? null
    );
  };

  return menuStructTree.map<CustomMenuItemType>((item) => {
    if (typeof item === 'string') {
      return getMenuItemWithKey(item);
    }

    if (item.type === 'group') {
      const children = item.group.map(getMenuItemWithKey);
      if (children.every((v) => v === null)) {
        return null;
      }
      return {
        type: 'group',
        label: item.label,
        children
      } as CustomMenuItemType;
    }

    return item as CustomMenuItemType;
  });
};
