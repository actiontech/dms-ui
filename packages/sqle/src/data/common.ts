export {
  ResponseCode,
  CharCode,
  SystemRole,
  ModalSize
} from '@actiontech/shared/lib/enum';

export {
  ModalFormLayout,
  PageBigFormLayout,
  PageFormLayout,
  filterFormButtonLayoutFactory,
  FilterFormColLayout,
  FilterFormLayout,
  FilterFormRowLayout
} from '@actiontech/shared/lib/data/common';

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export enum LoginTypeEnum {
  'ldap' = 'ldap',
  'sqle' = 'sqle'
}

export const SQLE_INSTANCE_SOURCE_NAME = 'SQLE';

export const SqlFiledInitialValue = '/* input your sql */';

export const SOURCE_WORKFLOW_PATH_KEY = 'sourceWorkflowId';

export const WORKFLOW_VERSION_NAME_PATH_KEY = 'versionName';

export const WORKFLOW_VERSION_ID_PATH_KEY = 'versionId';

export const SQL_MANAGEMENT_INSTANCE_PATH_KEY = 'instanceId';

export const SQL_MANAGEMENT_SOURCE_PATH_KEY = 'source';
