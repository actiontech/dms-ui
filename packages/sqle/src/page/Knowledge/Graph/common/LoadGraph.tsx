import {
  useLoadGraph,
  useRegisterEvents,
  useSetSettings,
  useSigma
} from '@react-sigma/core';
import { FC, useEffect } from 'react';
import useGraph from '../hooks/useGraph';
import { EdgeType, NodeType } from '../index.type';
import {
  INodeResponse,
  IEdgeResponse
} from '@actiontech/shared/lib/api/sqle/service/common';

type props = {
  graphData?: {
    nodes: INodeResponse[];
    edges: IEdgeResponse[];
  };
  hoveredNode: string | null;
  setHoveredNode: (node: string | null) => void;
};
const LoadGraph: FC<props> = ({ graphData, hoveredNode, setHoveredNode }) => {
  const { createGraph } = useGraph();
  const sigma = useSigma<NodeType, EdgeType>();
  const registerEvents = useRegisterEvents<NodeType, EdgeType>();
  const setSettings = useSetSettings<NodeType, EdgeType>();
  const loadGraph = useLoadGraph<NodeType, EdgeType>();

  useEffect(() => {
    if (graphData) {
      const graph = createGraph(graphData);
      loadGraph(graph);

      const handleEnterNode = (event: any) => {
        setHoveredNode(event.node);
      };

      const handleLeaveNode = () => {
        setHoveredNode(null);
      };

      registerEvents({
        enterNode: handleEnterNode,
        leaveNode: handleLeaveNode
      });

      return () => {
        // 清理事件监听
        sigma.removeAllListeners('enterNode');
        sigma.removeAllListeners('leaveNode');
      };
    }
  }, [
    graphData,
    createGraph,
    loadGraph,
    registerEvents,
    setHoveredNode,
    sigma
  ]);

  useEffect(() => {
    setSettings({
      nodeReducer: (node, data) => {
        const graph = sigma.getGraph();
        const newData = {
          ...data,
          highlighted: data.highlighted || false
        };

        if (hoveredNode && graph.hasNode(hoveredNode)) {
          try {
            const isNeighbor =
              node === hoveredNode ||
              graph.neighbors(hoveredNode).includes(node);
            if (isNeighbor) {
              newData.highlighted = true;
            } else {
              newData.color = '#E2E2E2';
              newData.highlighted = false;
            }
          } catch (error) {
            // 如果出现错误，保持节点原样
            console.debug('Failed to process node highlighting:', error);
          }
        }
        return newData;
      },
      edgeReducer: (edge, data) => {
        const graph = sigma.getGraph();
        const edgeSize = data.size ?? 3;
        const newData = { ...data, hidden: false, size: edgeSize };

        if (hoveredNode && graph.hasNode(hoveredNode)) {
          try {
            if (graph.extremities(edge).includes(hoveredNode)) {
              newData.size = edgeSize * 1.5;
            } else {
              newData.hidden = true;
            }
          } catch (error) {
            console.debug('Failed to process edge highlighting:', error);
          }
        }
        return newData;
      }
    });
  }, [hoveredNode, setSettings, sigma]);

  return null;
};

export default LoadGraph;
