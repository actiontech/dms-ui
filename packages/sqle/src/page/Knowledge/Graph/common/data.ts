import { SigmaContainerProps } from '@react-sigma/core';
import { EdgeType, NodeType } from '../index.type';
import { Attributes } from 'graphology-types';

export const sigmaSettings: SigmaContainerProps<
  NodeType,
  EdgeType,
  Attributes
>['settings'] = {
  // 允许在无效容器中渲染 (默认值: false)
  allowInvalidContainer: true,
  defaultNodeType: 'circle',
  // 设置边的默认类型 (默认值: 'line')
  defaultEdgeType: 'arrow',

  // 相机缩放比例限制 (默认值: min: 0.1, max: 32)
  minCameraRatio: 0.2,
  maxCameraRatio: 5,

  // 节点渲染程序类 (默认值: {})
  nodeProgramClasses: {},

  // 渲染性能优化
  hideEdgesOnMove: true, // 移动时隐藏边 (默认值: false)
  hideLabelsOnMove: false, // 移动时隐藏标签 (默认值: false)
  renderLabels: true, // 是否渲染标签 (默认值: true)
  renderEdgeLabels: false, // 是否渲染边标签 (默认值: false)

  // 标签渲染优化
  labelDensity: 0.1, // 标签密度 (默认值: 0.03)
  labelGridCellSize: 100, // 标签网格大小 (默认值: 100)
  labelRenderedSizeThreshold: 8, // 标签大小阈值 (默认值: 6)

  // 默认样式
  defaultNodeColor: '#1a73e8', // 节点颜色 (默认值: '#999')
  defaultEdgeColor: '#ccc', // 边颜色 (默认值: '#ccc')

  // 交互体验优化
  zoomToSizeRatioFunction: (x: number) => x // 缩放比例函数 (默认值: Math.pow)
};
