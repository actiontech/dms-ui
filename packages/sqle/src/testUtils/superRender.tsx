import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
