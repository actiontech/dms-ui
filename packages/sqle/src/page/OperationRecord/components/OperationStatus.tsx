import React from 'react';
import { useTranslation } from 'react-i18next';
import { OperationStatusStyleWrapper } from '../style';
import { OperationRecordListStatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';
import {
  IconOrderStatusIsFinished,
  IconOrderStatusIsFailed
} from '../../../icon/Order';

const OperationStatus: React.FC<{ status: OperationRecordListStatusEnum }> = ({
  status
}) => {
  const { t } = useTranslation();

  return (
    <OperationStatusStyleWrapper>
      {status === OperationRecordListStatusEnum.succeeded ? (
        <>
          <IconOrderStatusIsFinished />
          {t('common.success')}
        </>
      ) : (
        <>
          <IconOrderStatusIsFailed />
          {t('common.fail')}
        </>
      )}
    </OperationStatusStyleWrapper>
  );
};

export default OperationStatus;
