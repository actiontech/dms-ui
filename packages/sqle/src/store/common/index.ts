import { ModalStatus } from '@actiontech/dms-kit/es/types/common.type';
import { PayloadAction } from '@reduxjs/toolkit';

export const commonModalReducer = <
  T extends { modalStatus: ModalStatus }
>() => ({
  initModalStatus(
    state: T,
    { payload: { modalStatus } }: PayloadAction<{ modalStatus: ModalStatus }>
  ) {
    state.modalStatus = modalStatus;
  },
  updateModalStatus(
    state: T,
    {
      payload: { modalName, status }
    }: PayloadAction<{ modalName: string; status: boolean }>
  ) {
    if (Object.prototype.hasOwnProperty.call(state.modalStatus, modalName)) {
      state.modalStatus[modalName] = status;
    }
  }
});
