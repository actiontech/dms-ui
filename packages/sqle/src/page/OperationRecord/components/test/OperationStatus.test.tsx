import OperationStatus from '../OperationStatus';
import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import { OperationRecordListStatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

describe('sqle/OperationRecord/OperationStatus', () => {
  it('should match snap shot', () => {
    const { baseElement } = superRender(
      <OperationStatus status={OperationRecordListStatusEnum.failed} />
    );
    const { baseElement: baseElement2 } = superRender(
      <OperationStatus status={OperationRecordListStatusEnum.succeeded} />
    );
    expect(baseElement).toMatchSnapshot();
    expect(baseElement2).toMatchSnapshot();
  });
});
