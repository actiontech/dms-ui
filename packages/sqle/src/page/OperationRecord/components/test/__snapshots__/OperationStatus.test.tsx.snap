// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/OperationRecord/OperationStatus should match snap shot 1`] = `
<body>
  <div>
    <div
      class="css-10kos8e"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949zM8 10v2M8 4l.004 4.5"
          stroke="#F59957"
          stroke-width="1.5"
        />
      </svg>
      <span>
        失败
      </span>
    </div>
  </div>
  <div>
    <div
      class="css-10kos8e"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949z"
          stroke="#41BF9A"
          stroke-width="1.5"
        />
        <path
          d="m5.083 8 2.084 2.083 4.166-4.166"
          stroke="#41BF9A"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <span>
        成功
      </span>
    </div>
  </div>
</body>
`;

exports[`sqle/OperationRecord/OperationStatus should match snap shot 2`] = `
<body>
  <div>
    <div
      class="css-10kos8e"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949zM8 10v2M8 4l.004 4.5"
          stroke="#F59957"
          stroke-width="1.5"
        />
      </svg>
      <span>
        失败
      </span>
    </div>
  </div>
  <div>
    <div
      class="css-10kos8e"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949z"
          stroke="#41BF9A"
          stroke-width="1.5"
        />
        <path
          d="m5.083 8 2.084 2.083 4.166-4.166"
          stroke="#41BF9A"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <span>
        成功
      </span>
    </div>
  </div>
</body>
`;
